app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,69 @@
+class PostsController < ApplicationController
+  before_action :set_post, only: %i[show edit update destroy]
+  before_action :authenticate_user!, only: %i[new create]
+
+  # GET /posts
+  # GET /posts.json
+  def index
+    @posts = Post.all
+  end
+
+  # GET /posts/1
+  # GET /posts/1.json
+  def show; end
+
+  # GET /posts/new
+  def new
+    @post = Post.new
+  end
+
+  # GET /posts/1/edit
+  def edit; end
+
+  # POST /posts
+  # POST /posts.json
+  def create
+    @post = Post.new(title: params[:post][:title], body: params[:post][:body], user_id: current_user.id)
+
+    respond_to do |format|
+      if @post.save
+        format.html { redirect_to posts_path, notice: 'Post was successfully created.' }
+        format.json { render :show, status: :created, location: @post }
```

Comment:
Is there a specific reason you are rendering the json file?? It's not necessary if you are using the html views.
If you used the scaffold method to create your app (I don't suggest you use it anymore) these lines are created automatically but that is useful in other scenarios that you will see in the future.